snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/internal/emitter/Emitter.java

```diff
@@ -530,7 +531,7 @@ private void attemptEmit() {
     protected List<Request> buildRequests(@NonNull List<EmitterEvent> events) {
         List<Request> requests = new ArrayList<>();
         String sendingTime = Util.getTimestamp();
-        HttpMethod httpMethod = networkConnection.getHttpMethod();
+        HttpMethod httpMethod = getNetworkConnection().getHttpMethod();
```

Comment:
Maybe we could pass `httpMethod` as argument of `buildRequests`?
I'm thinking about the case where `buildRequests` is called inside `attemptEmit` but meanwhile the networkConnection changes. Apparently `buildRequests` and `isOversize` just need the httpMethod.